aws/resource_aws_vpc_endpoint.go

```diff
@@ -58,180 +90,295 @@ func resourceAwsVpcEndpoint() *schema.Resource {
 				Computed: true,
 				Elem:     &schema.Schema{Type: schema.TypeString},
 			},
+			"network_interface_ids": {
+				Type:     schema.TypeSet,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Set:      schema.HashString,
+			},
+			"dns_entry": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"dns_name": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"hosted_zone_id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
 		},
 	}
 }
 
-func resourceAwsVPCEndpointCreate(d *schema.ResourceData, meta interface{}) error {
-	conn := meta.(*AWSClient).ec2conn
-	input := &ec2.CreateVpcEndpointInput{
-		VpcId:       aws.String(d.Get("vpc_id").(string)),
-		ServiceName: aws.String(d.Get("service_name").(string)),
+func resourceAwsVpcEndpointCreate(d *schema.ResourceData, meta interface{}) error {
+	if d.Get("vpc_endpoint_type").(string) == ec2.VpcEndpointTypeInterface &&
```

Comment:
I think we can now replace this "apply-time" logic with "plan-time" logic using [these changes](https://github.com/hashicorp/terraform/pull/14887).

Reply:
I can't get the "plan-time" logic
```
CustomizeDiff: func(diff *schema.ResourceDiff, v interface{}) error {
    if diff.Get("vpc_endpoint_type").(string) == ec2.VpcEndpointTypeInterface {
        _, new := diff.GetChange("security_group_ids")
        if new.(*schema.Set).Len() == 0 {
                return errors.New("An Interface VPC Endpoint must always have at least one Security Group")
        }
    }

    return nil
},
```
to work - the new length is always 0 when creating the resource the first time.